src/server/pachyderm_test.go

```diff
@@ -10634,7 +10635,14 @@ func TestDatabaseStats(t *testing.T) {
 				}
 			}
 
-			require.Equal(t, true, commitsFound, "we should have an entry in row-counts.json for commits")
+			require.Equal(t, true, commitsFound,
+				"we should have an entry in row-counts.json for commits")
+		}
+
+		if hdr.Name == "database/tables/collections/commits.json" {
+			require.NoError(t, json.Unmarshal(fileContents.Bytes(), &rows),
+				"unmarshalling commits.json should succeed")
+			require.Equal(t, numCommits, len(rows), "number of commits should match number of rows")
 		}
```

Comment:
This test will still pass if given an empty tar file.  You need to keep some state like `var foundRowCounts, foundCommitsJSON`, set them to true in each branch, and check at the end of the test that they're true.  You don't want to test "if there's a file called database/row-counts.json it's valid", you want to test "there must be a file called database/row-counts.json and it must be valid".